docs/pages/deploy-a-cluster/teleport-enterprise/gcp-kms.mdx

```diff
@@ -0,0 +1,270 @@
+---
+title: Google KMS
+description: Configure Teleport to store CA private keys in the Google Cloud Key Management Service
+h1: Store Teleport Private Keys in Google Cloud KMS
+---
+
+This guide will show you how to set up your Teleport Cluster to use the Google
+Cloud Key Management Service (KMS) to store and handle the CA private key
+material used to sign all certificates issued by your Teleport cluster.
+
+Teleport generates User, Host, JWT, and Database Certificate Authorities (CAs)
+during the first Auth Server's initial startup.
+These CAs are used to sign all certificates issued to clients and hosts in the
+Teleport cluster.
+When configured to use Google KMS, all private key material for these CAs will
+be generated, stored, and used for signing inside of Google KMS.
+In short, private key material will never leave Google KMS.
+Instead of the actual private key, Teleport will only store the ID of the KMS
+key.
+
+If launching a new Teleport cluster this will all be handled during initial
+startup with no specific interventions required after configuration. If
+configuring an existing Teleport cluster to switch to KMS keys after it has
+already started with software keys a CA rotation must be performed, read on to
+[migrating an existing cluster](#migrating-an-existing-cluster) to learn more.
+
+<ScopedBlock scope={["oss", "cloud"]}>
+
+This guide is intended for self-hosted Teleport Enterprise users.
+
+</ScopedBlock>
+
+## Prerequisites
+
+- Teleport v(=teleport.version=) Enterprise (self-hosted).
```

Comment:
I've never been a fan of using `v(=teleport.version=)` in the prerequisites section, as the docs always read like you need the latest version in order to use the feature.

Wouldn't it be best if the prerequisites for the feature said "Teleport version X or newer" where X is the version where we introduced the feature? Without this, it's hard to know if things aren't working because there's a misconfiguration or because the version of Teleport is too old.

Reply:
I agree that's better. Was told to do it this way at some point in the past and have copied it ever since. I've updated this to say 11.1 for now and will correct with the real version in backports